src/main/java/trade/wayruha/mexc/ws/WebSocketClient.java

```diff
@@ -72,77 +76,103 @@ public class WebSocketClient<T> extends WebSocketListener {
     }
 
-    public void send(String str) {
+    public void sendRequest(String request) {
         boolean result = false;
-        log.debug("{}:send() {}", logPrefix, str);
+        log.debug("{} Try to send request {}", logPrefix, request);
         if (webSocket != null) {
-            result = webSocket.send(str);
+            result = webSocket.send(request);
         }
         if (!result) {
-            log.error("{}: Failed to send message: {}", logPrefix, str);
+            log.error("{} Failed to send message: {}", logPrefix, request);
             closeOnError(null);
         }
     }
 
     @SneakyThrows
     public void subscribe(Collection<String> newChannels) {
-        if (this.channels.size() + newChannels.size() > WEB_SOCKET_MAX_CHANNELS_PER_CONNECTION)
-            throw new IllegalArgumentException("Can't subscribe to more then " + WEB_SOCKET_MAX_CHANNELS_PER_CONNECTION + "channels in single WS connection");
+        if (this.channels.size() + newChannels.size() > this.config.getWebSocketMaxReconnectAttemptNumber())
+            throw new IllegalArgumentException("Can't subscribe to more then " + WEB_SOCKET_MAX_CHANNELS_PER_CONNECTION + " channels in single WS connection");
         final Message msg = new Message(Action.SUBSCRIPTION, newChannels);
         final String content = objectMapper.writeValueAsString(msg);
-        this.send(content);
+        this.sendRequest(content);
         this.channels.addAll(newChannels);
     }
 
     @SneakyThrows
     public void ping(Collection<String> channels) {
         final Message msg = new Message(Action.PING, channels);
         final String content = objectMapper.writeValueAsString(msg);
-        this.send(content);
-        log.debug("{} ping was send to channels {} ", logPrefix, channels);
+        this.sendRequest(content);
+        log.trace("{} Ping was send to channels {} ", logPrefix, channels);
     }
 
     @SneakyThrows
     public void unsubscribe(Collection<String> channels) {
         final Message msg = new Message(Action.UNSUBSCRIPTION, channels);
         final String content = objectMapper.writeValueAsString(msg);
-        send(content);
+        sendRequest(content);
         this.channels.removeAll(channels);
     }
 
     public void close() {
-        log.debug("{} closing", logPrefix);
+        log.debug("{} Closing WS channels: {}", logPrefix, this.channels);
+        state = WSState.IDLE;
         if (webSocket != null) {
             webSocket.cancel();
             webSocket = null;
         }
-        state = WSState.IDLE;
         this.channels.clear();
         scheduledPingTask.cancel(false);
     }
 
-    public void reConnect() {
-        this.close();
-        this.connect(this.channels);
+    public boolean reConnect() {
+        var success = false;
+        this.canceledChannels = new HashSet<>();
+        this.canceledChannels.addAll(this.channels);
```

Comment:
shorter form: this.canceLLedChannels = new HashSet<>(this.channels) ?

Reply:
:+1: 
fixed